components/script/dom/document.rs

```diff
@@ -2784,9 +2797,60 @@ impl Document {
             shadow_roots: DomRefCell::new(HashSet::new()),
             shadow_roots_styles_changed: Cell::new(false),
             media_controls: DomRefCell::new(HashMap::new()),
+            csp_list: DomRefCell::new(None),
+        }
+    }
+
+    pub fn set_csp_list(&self, csp_list: Option<CspList>) {
+        *self.csp_list.borrow_mut() = csp_list;
+    }
+
+    pub fn get_csp_list(&self) -> Option<Ref<CspList>> {
+        let b = self.csp_list.borrow();
+        if b.is_none() {
+            None
+        } else {
+            Some(Ref::map(b, |o| o.as_ref().unwrap()))
```

Comment:
Use [`ref_filter_map`](https://docs.rs/ref_filter_map/1.0.1/ref_filter_map/fn.ref_filter_map.html). Servo already depends on it anyway.